dmscripts/generate_framework_agreement_signature_pages.py

```diff
@@ -95,28 +108,65 @@ def render_html_for_suppliers_awaiting_countersignature(rows, framework, templat
         data['countersigned_at'] = datetime.strptime(
             data['countersigned_at'], '%Y-%m-%dT%H:%M:%S.%fZ'
         ).strftime('%d %B %Y')
+        if data['signed_agreement_returned_at']:
+            data['signed_agreement_returned_at'] = datetime.strptime(
+                data['signed_agreement_returned_at'], '%Y-%m-%dT%H:%M:%S.%fZ'
+            ).strftime('%d %B %Y')
         data['includeCountersignature'] = True
-        html = render_html(template_path, data)
-        save_page(html, data['supplier_id'], output_dir, "agreement-countersignature")
-    shutil.copyfile(template_css_path, os.path.join(output_dir, 'framework-agreement-signature-page.css'))
-    shutil.copyfile(
-        template_countersignature_path,
-        os.path.join(output_dir, 'framework-agreement-countersignature.png')
-    )
 
+        html_dir = output_dir / str(data['supplier_id'])
+        output_pages = []
+        for html_page in html_pages:
+            html = render_html(str(html_page), data)
+            page_path = save_page(html, data['supplier_id'], html_dir, html_page.stem)
+            output_pages.append(Path(page_path).resolve())
+
+        for static_file in static_files:
+            shutil.copyfile(static_file, Path(html_dir, static_file.name))
+
+        yield html_dir, output_pages
+
+
+def render_pdf_for_each_html_page(html_pages, html_dir, pdf_dir, framework_slug):
+    html_dir = Path(html_dir).resolve()
+    pdf_dir = Path(pdf_dir).resolve()
 
-def render_pdf_for_each_html_page(html_pages, html_dir, pdf_dir):
-    html_dir = os.path.abspath(html_dir)
-    pdf_dir = os.path.abspath(pdf_dir)
     ok = True
+
     if not os.path.exists(pdf_dir):
         os.mkdir(pdf_dir)
-    for index, html_page in enumerate(html_pages):
-        html_path = os.path.join(html_dir, html_page)
-        pdf_path = '{}'.format(re.sub(r'\.html$', '.pdf', html_path))
-        pdf_path = '{}'.format(re.sub(html_dir, pdf_dir, pdf_path))
+
+    if len(html_pages) > 1:
+        for html_path in html_pages:
+            pdf_path = html_dir / (html_path.stem + ".pdf")
+            exit_code = subprocess.call(['wkhtmltopdf', 'file://{}'.format(html_path), pdf_path])
```

Comment:
> `wkhtmltopdf`

Do we need to document this as a requirement of this script?

Reply:
@bjgill This is documented in the parent script used to call this function: https://github.com/alphagov/digitalmarketplace-scripts/blob/0274a566c88a185833899523870b78c35382cd87/scripts/generate-framework-agreement-counterpart-signature-pages.py#L4-L5